hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -84,8 +110,20 @@ def _strategy_for(param: inspect.Parameter) -> Union[st.SearchStrategy, InferTyp
     if param.annotation is not inspect.Parameter.empty:
         return infer
     # If there's no annotation and no default value, the user will have to
-    # fill this in later.  We use nothing() as a placeholder to this effect.
+    # fill this in later.  We use nothing() as a placeholder to this effect,
+    # if none of our name-based heuristics are applicable.
     if param.default is inspect.Parameter.empty:
+        if "string" in param.name:
+            return st.text()
+        for strategy, names in [
+            (st.text(), ["name", "filename", "fname"]),
+            (st.integers(min_value=0), ["index"]),
+            (st.floats(), ["real", "imag"]),
+            (st.functions(), ["function", "func", "f"]),
+            (st.iterables(st.integers()) | st.iterables(st.text()), ["iterable"]),
```

Comment:
It seems like it would be worth it to make this a proper module-level dictionary so that it is easy to reference / understand some of the decisions that ghostwriter is making